classes/image-adbd.bbclass

```diff
@@ -0,0 +1,13 @@
+# This class installs adbd into the target image
+# It is disabled by default unless either 'start-adbd' or 'debug-tweaks' is
+# present in "IMAGE_FEUTURES"
+
+PACKAGE_INSTALL_append = " android-tools-adbd "
+
+IMAGE_FEATURES[validitems] += "start-adbd"
+
+start_adbd_at_boot () {
+	touch ${IMAGE_ROOTFS}/var/usb-debugging-enabled
```

Comment:
How it works?, is systemd adb unit file looks for this file to be enabled?.

Reply:
@alimon yes, adbd systemd service has a condition to check for this file:

https://github.com/openembedded/meta-openembedded/blob/master/meta-oe/recipes-devtools/android-tools/android-tools/android-tools-adbd.service

Reply:
don't we need to set PREFERRED_PROVIDER for android-tools-conf somewhere?

also, I am wondering if adding adbd support is really an image config, as opposed to a machine config.. what do you think?

Reply:
There is one default in the meta-oe's `conf/layer.conf`:
> PREFERRED_PROVIDER_android-tools-conf ?= "android-tools-conf"

Machine config can override this default (see https://github.com/ndechesne/meta-qcom/pull/227).

I suppose that adbd is an image feature rather than a machine config: consider one enabling adbd for all `rpb-foo-test` images only, but disabling it for production images.